routes/ecom/modules/list-payments.js

```diff
@@ -3,12 +3,40 @@
 // generate default payment gateway object
 const newPaymentGateway = require(process.cwd() + '/lib/new-payment-gateway')
 
+const checkDiscount = (paymentGateway, items) => {
+  const discount = paymentGateway.discount
+  const productIds = discount?.product_ids
+  if (items && items.length) {
```

Comment:
Um early return caberia bem aqui:

```suggestion
  if (!productIds || !productIds.length) return
  if (items && items.length) {
```

Evitaria ficar repetindo esse `productIds && productIds.length` e só segue na função já sabendo que `productIds.length > 0`, que é único cenário que interessa aqui mesmo...